Userland/Libraries/LibGUI/DynamicWidgetContainer.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023, Torsten Engelmann <engelTorsten@gmx.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibGUI/Frame.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/Window.h>
+
+namespace GUI {
+
+class DynamicWidgetContainer : public Frame {
+    C_OBJECT(DynamicWidgetContainer);
+
+public:
+    enum class ViewState {
+        Expanded,
+        Collapsed,
+        Detached
+    };
+
+    enum class MoveTargetOperation {
+        SetTarget,
+        ClearAllTargets
+    };
+
+    ViewState view_state() const { return m_view_state; }
+    void set_view_state(ViewState);
+    StringView section_label() const& { return m_section_label; }
+    void set_section_label(String);
+    StringView config_domain() const& { return m_config_domain; }
+    void set_config_domain(String);
+    bool persist_state() const { return !m_config_domain.is_empty(); }
+    void set_detached_size(Gfx::IntSize const);
+    Gfx::IntSize detached_size() const { return m_detached_size.value(); }
+    void set_container_with_individual_order(bool);
+    bool is_container_with_individual_order() const { return m_is_container_with_individual_order; }
+    void set_show_controls(bool);
+    bool show_controls() const { return m_show_controls; }
+    void set_parent_container(RefPtr<GUI::DynamicWidgetContainer>);
+    bool check_has_move_target(Gfx::IntPoint, MoveTargetOperation);
+    // FIXME: this should not be a public api and static - but currently the destructor is not being called when the widget was created via a gml file.
```

Comment:
:thinking: Is this true for all widgets created from GML? If you can find a reduced test case it would be useful to create an issue about this. (Even if you can't reduce it, create an issue with whatever you know.)